Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -31,38 +34,39 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
     /// This pool's event loop.
     public let eventLoop: EventLoop
     
+    /// ID generator
+    private var idGenerator = ManagedAtomic(0)
```

Comment:
Should this be a static var for ids to be unique across all connection pools?
Or should the connection pool have its own id, which we can attach to the logger in init?

Reply:
These IDs are used solely as keys in the `OrderedDictionary` for easy removal of waiters when they time out or are otherwise succeeded or failed by something other than being popped off the list. There's no need for it to be shared and the only reason to log it is for tracing the operation of the pool itself.

Reply:
The old implementation basically used the address of `promise.futureResult` as the hash key (except it didn't even bother using a hash table at all), but that's long been considered very poor form and subject to unpredictable issues, so I switched to a stable monotonically increasing integer.

Reply:
it should be `private let` though?

Reply:
Ah, good catch!

Reply:
Also, how can you actually trace if there are 2 different connection pools that somehow log the same waiter id at the same time? Are you sure that won't be a problem?

Reply:
Oh, that's pretty much guaranteed to happen if there's more than one event loop in a given Vapor app. It doesn't matter; the IDs don't actually mean anything and deciphering them in log messages isn't intended to be something users should ever need to do. Even treating them as a hint as to the order of queries is arguably wrong, the same way that an `INTEGER PRIMARY KEY AUTO_INCREMENT` column in a MySQL database can't be assumed to have IDs that signify a meaningful ordering.

Reply:
My concern isn't order, it's just the uniqueness of logs.
Imagine someone seeing 
```
"Adding connection request to waitlist with ID 1029"
"Adding connection request to waitlist with ID 1029"
"Adding connection request to waitlist with ID 1029"
```
And thinking what's up with the connection pool.

It'll be rare so it's not a big deal but I'm not convinced a shared counter isn't better so we can make sure chances of something like this happening are eliminated 🤔

Reply:
@gwynne no conversation resolution for you yet 😝

Reply:
It's guaranteed to happen and the ID isn't expected or intended to be useful to those looking at the logs. I only bothered to log it at all while I was debugging the bizarre ordering dependencies of the existing tests.